mautic/mautic
# Set secure cookie from Request when in installer and fixes configuration for new installation

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
The installer does not have access to the site_url because it has not been written to the config yet. So this change checks for https from the Request if in the installer to know whether to set secure cookie. 

It also removes advice to install APC during the installation since that is outdated.

Finally, it fixes https://github.com/mautic/mautic/issues/8423 which makes the configuration inaccessible for new installations.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Install Mautic 3.x through https
2. Look at the cookies in the developer console and notice that the PHP session cookie is not marked as Secure.
3. Go to Mautic's Configuration and you'll be greeted by an error page.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Run through the installer in https://
3. Cookies should be set to secure 
4. Go to Mautic's Configuration and it should be accessible
5. Clear the browser's cache for at least the Mautic site
6. Install Mautic again under http://
7. This time the session cookie should not be set as Secure
